updater/lib/dependabot/dependency_change_builder.rb

```diff
@@ -28,7 +28,12 @@ def self.create_from(**kwargs)
 
     def initialize(job:, dependency_files:, updated_dependencies:, change_source:)
       @job = job
-      @dependency_files = dependency_files
+
+      dir = Pathname.new(job.source.directory).cleanpath
+      @dependency_files = dependency_files.select { |f| Pathname.new(f.directory).cleanpath == dir }
+
+      raise "Missing directory in dependency files: #{dir}" unless @dependency_files.any?
```

Comment:
Should we have a well defined exception here? That way it's easier to track in Sentry and other places?

Reply:
Raising like this will result in a RuntimeError exception. Just checked and we have 161 places in the code that raises in the same way. 

For unique scenarios and sanity checks I think raising like this is ok. It would quickly become unmanageable to create a new exception class for each one. Sentry is good about grouping by stack trace and not the exception class name so I don't think it will make a difference, just more code to manage.

Reply:
When exactly does this happen @jakecoffman?

Reply:
If the file_fetcher didn't set the directory (meaning the ecosystem didn't use the `fetch_file_if_present`), or the directory wasn't cleaned and so the comparison fails.

It's more of a sanity check. It happened once while I was working on this PR and resulted in a misleading error so I added this raise to point us in the right direction if it happens again.